### ISSUE TITLE
Configure fails on some windows machines after #22097

### ISSUE THREAD

**Issue by davecrighton** [openssl/openssl#23591]
openSSL 3.0.13 fails to build on windows with the following error:

```
11:27:21       [exec] Configuring OpenSSL version 3.0.13+quic for target VC-WIN64A
11:27:21       [exec] Using os-specific seed configuration
11:28:15       [exec] 
11:28:15       [exec] Failure!  makefile wasn't produced.
11:28:15       [exec] Please read INSTALL.md and associated NOTES-* files.  You may also have to
11:28:15       [exec] look over your available compiler tool chain or change your configuration.
11:28:15       [exec] 
11:28:15       [exec] C:\ci\obj\openssl\crypto\providers: No such file or directory at Configure line 3416.
```

This doesnt happen on all our windows system but does happen on our pipeline build servers. Looking into it further the affected system does not have a read only source tree and after the build the listed directory does not exist.

Running on another windows system this directory appears to be created during the build.

I am running `perl Configure VC-WIN64A`  and backing out part of the change to Configure made under https://github.com/openssl/openssl/issues/22907 resolves the issue, the specific changes I made were to comment out the added `unless $nomkpath` added to  the `cleandir` and `cleanfile` subroutines. ie/

```
sub cleandir {
        my $base = shift;
        my $dir = shift;
        my $relativeto = shift || ".";
        my $no_mkpath = shift // 0;

        $dir = catdir($base,$dir) unless isabsolute($dir);

        # Make sure the directories we're building in exists
        mkpath($dir); // unless $no_mkpath;

        my $res = abs2rel(absolutedir($dir), rel2abs($relativeto));
        #print STDERR "DEBUG[cleandir]: $dir , $base => $res\n";
        return $res;
    }

    sub cleanfile {
        my $base = shift;
        my $file = shift;
        my $relativeto = shift || ".";
        my $no_mkpath = shift // 0;

        $file = catfile($base,$file) unless isabsolute($file);

        my $d = dirname($file);
        my $f = basename($file);

        # Make sure the directories we're building in exists
        mkpath($d); // unless $no_mkpath;

        my $res = abs2rel(catfile(absolutedir($d), $f), rel2abs($relativeto));
        print STDERR "DEBUG[cleanfile]: $d , $f => $res\n";
        return $res;
    }
```

**Comment by t8m** [comment]
Duplicate of #23436
